server.go

```diff
@@ -26,6 +26,11 @@ var (
 	defaultTimeout     = 10 * time.Second
 )
 
+type ConnectionStateChange struct {
+	Identifier string
+	Connected  bool
```

Comment:
`StateChange` is about state transition, this value carries information how the connection state transitioned. While I don't see for now use of tracking connection's previous state, can we add `Current ServerState` field here and get rid of `Connected` one? Then we could have `ServerConnected`, `ServerDisconnected` and in future it would be easier to add more fine-grained events like `ServerSessionRequested`, `ServerHeartbeat`, `ServerSessionRetry` etc. Something like:

```go
type ServerState uint8

type ServerStateChange struct {
   Identifier string
   Current  ServerState
}
```

I was also thinking how hard would it be to add events for abnormal disconnections - e.g. yamux transport timed out sending heartbeat or requesting session and decided to drop connection; for such cases emitting `ServerStateChange` with `Error error` field set would be handful for e.g. tracking statistics. Just thinking loudly.